src/screens/ChangeWalletSheet.js

```diff
@@ -109,6 +115,10 @@ export default function ChangeWalletSheet() {
     showDividers = true;
   }
 
+  useEffect(() => {
+    setCurrentAddress(accountAddress);
```

Comment:
Can you elaborate on why it actually fixes anything?

Reply:
In every other scenario, currentAddress is set during wallet change and it works perfectly well because it doesn't close the change wallet modal.
Wallet Creation is different because it opens a new` CreationWallet modal`, and close `ChangeWallet modal` at the same time. This means all `setCurrentWallet()` hooks send to `CreationWallet modal` won't affect `ChangeWallet` when it reappears, but it will try to change things in the old one that's already closed. 
But `CreationWallet` also change the accountAddress variable in `useAccountAddress` and since `ChangeWallet`use this hook we can simply useEffect on every accountAddress change to make it currentAddress that checkmark is based on.

It's worth to leave all `setCurrentWallet()` though because in other scenarios setting checkmark manually is faster and more responsive than waiting for useAccountAddress hook to update